crates/refunder/src/refund_service.rs

```diff
@@ -1,47 +1,135 @@
-use anyhow::Result;
-use database::ethflow_orders::refundable_orders;
+use anyhow::{anyhow, Result};
+use contracts::CoWSwapEthFlow;
+use database::{
+    ethflow_orders::{mark_eth_orders_as_refunded, refundable_orders, EthOrderPlacement},
+    OrderUid,
+};
+use primitive_types::H160;
+use shared::{transport::MAX_BATCH_SIZE, Web3, Web3CallBatch};
 use sqlx::PgPool;
 use std::time::SystemTime;
 
+const INVALIDATED_OWNER: H160 = H160([255u8; 20]);
+
 pub struct RefundService {
     pub db: PgPool,
+    pub web3: Web3,
+    pub ethflow_contract: CoWSwapEthFlow,
     pub min_validity_duration: i64,
     pub min_slippage: f64,
 }
 
+type SplittedOrderUids = (Vec<(OrderUid, bool)>, Vec<(OrderUid, bool)>);
```

Comment:
It seems a bit error prone that we don't know from the type which side is the refunded one and which is the non-refunded one. Maybe this could be made into a struct instead for clarity?